core/base/src/TDirectory.cxx

```diff
@@ -1297,12 +1351,29 @@ void TDirectory::DecodeNameCycle(const char *buffer, char *name, Short_t &cycle,
       cycle = 9999;
 }
 
+void TDirectory::TContext::RegisterCurrentDirectory()
+{
+   // peg the current directly
+   TDirectory *current;
+   {
+      ROOT::Internal::TSpinLockGuard slg(*GetCurrentDirectoryLock());
+      current = TDirectory::CurrentDirectory().load();
+      if (!current || !current->IsBuilt())
```

Comment:
Why should we not add a half-built `TDirectory` so e.g. `~TContext()` can return to it? (An explanatory comment would be appreciated.)

Reply:
If it was not fully built it would not (yet) be a `gDirectory`.  What this captures is that the directory is in the process of being destructed.  I.e. we reached here while another thread had already started to run the destructor but not yet cleared this thread `gDirectory` (likely be the lock that this thread holds is preventing/halting the completion of that destructor.